operational_analysis/methods/plant_analysis.py

```diff
@@ -802,13 +846,50 @@ def filter_outliers(self, n):
             value_min=0.02 * plant_capac,
             value_max=1.2 * plant_capac,
         )
-
+        
+        if self.outlier_detection:
+            if self.time_resolution == 'M':
+                # Monthly linear regression (i.e., few data points): 
+                # filter outliers based on robust linear regression
+                # using Huber algorithm to flag outliers
+                X = sm.add_constant(df_sub[reanal])  # Reanalysis data with constant column
+                y = df_sub['gross_energy_gwh']*30/df_sub['num_days_expected']  # Energy data
+        
+                # Perform robust linear regression
+                rlm = sm.RLM(y, X, M=sm.robust.norms.HuberT(self._run.outlier_threshold))
+                rlm_results = rlm.fit()
+        
+                # Define valid data as points in which the Huber algorithm returned a value of 1
+                df_sub.loc[:, "flag_outliers"] = rlm_results.weights != 1
+            
+            else:
+                # Daily regressions (i.e., higher number of data points):
+                # Apply bin filter to catch outliers
+                df_sub.loc[:, "flag_outliers"] = filters.bin_filter(
```

Comment:
For daily/hourly resolution, filtering is performed using wind speed only. But the ML methods could use wind direction and temperature as additional predictor variables, in part to account for some of the variability in energy for a given wind speed. So does it make sense to only filter using the wind speed/energy curve? Or perhaps discuss this assumption/limitation in the comments or documentation somewhere.

Reply:
Yes, this is a good point. The reason for not applying any filters based on different variables was that there's not an obvious relationship between wind direction (or temperature) and power, so it would be hard to come up with filters there. The same is missing in the TIE calculation, too. Maybe something to keep in mind for a future addition?